WordPress/build.gradle

```diff
@@ -155,6 +155,7 @@ dependencies {
     testImplementation 'org.robolectric:shadows-multidex:3.6.1'
     testImplementation 'org.mockito:mockito-core:2.8.9'
     testImplementation 'com.nhaarman:mockito-kotlin:1.5.0'
+    testImplementation 'nl.jqno.equalsverifier:equalsverifier:2.5.2'
```

Comment:
This bug surfaced when the field `isPortfolioEnabled` and `portfolioPostsPerPage` were added to the `equals` method. It turned out that the bug needed to be fixed for the change to work. That's how the commit 885d7fa actuallly made it as part of the "introduce portfolio" PR, and actually relates to the feature being implemented here. I hope it makes sense.

To your second point (introducing the library `equalsverifier:equalsverifier ` for testing), it totally makes sense - still learnig how to contribute to wordpress-mobile! I've also been thinking since then, that it's not even useful to test it if we can convert this class to a Kotlin `data class`. 

Because, it's a model I think it would be a good fit for a `data class`, it would also fix the equals issue, even remove the need of testing it and, enables us to introduce the fields `isPortfolioEnabled` and `portfolioPostsPerPage`.

Does it seem reasonable to you @mzorz ?

Reply:
I went ahead and converted `SiteSettingsModel` to a Kotlin class for the aforementioned reasons. https://github.com/wordpress-mobile/WordPress-Android/pull/8221/commits/ef69d35087470521c4b50724b727b740ff054d2c

Please, let me know if you're fine with it.

Reply:
> Because, it's a model I think it would be a good fit for a data class, it would also fix the equals issue, even remove the need of testing it and, enables us to introduce the fields isPortfolioEnabled and portfolioPostsPerPage.

Nice move! - I was confused at first as I wasn't expecting  the solution to be like this :D - but I realize this aligns with our goal to move to Kotlin 1 piece at a time as we find the opportunity 
 (you don't need to know, this is just an extra thing), and you happened to have found one of these opportunities. Good one! 🙌